### PR TITLE
Base container

### PR DESCRIPTION
**Pull request by drivera-armedia**
First version of the container to be merged to the main branch

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,32 @@
...
+LABEL maintainer="Armedia, LLC"
 
+ARG ARCH="amd64"
+ARG OS="linux"
+ARG VER="0.22.2"
+ARG PKG="alertmanager"
+ARG SRC="${PKG}-${VER}.${OS}-${ARCH}"
+
+RUN curl \
+		-L "https://github.com/prometheus/${PKG}/releases/download/v${VER}/${SRC}.tar.gz" \
```
Comment (JonHolman) [inline Dockerfile]: @drivera-armedia  please download this artifact to our artifacts s3 bucket "arkcase-container-artifacts" and add a script like this to your PR that CodeBuild will run prior to building the dockerfile https://github.com/ArkCase/ark_activemq/blob/fabrice-debase/get-artifacts.sh

File: Dockerfile [hunk 2]
```diff
@@ -1 +1,32 @@
+ARG ARCH="amd64"
+ARG OS="linux"
```
Comment (JonHolman) [inline Dockerfile]: Why are these two ARGs in here twice?

File: Dockerfile [hunk 3]
```diff
@@ -1 +1,32 @@
...
+FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest
+LABEL maintainer="Armedia, LLC"
```
Comment (JonHolman) [inline Dockerfile]: please include more labels, like:
LABEL ORG="Armedia LLC" \
    APP="Samba" \
    VERSION="1.0" \
    IMAGE_SOURCE="https://github.com/ArkCase/ark_samba" \
    MAINTAINER="Armedia LLC"

**Review by JonHolman — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -1 +1,32 @@
...
+		-o package.tar.gz && \
+	tar -xzvf "package.tar.gz" && \
+	mkdir -pv "/etc/alertmanager" && \
+	mkdir -p "/alertmanager" && \
+	mv -vif "${SRC}/LICENSE"                     "/LICENSE" && \
+	mv -vif "${SRC}/NOTICE"                      "/NOTICE" && \
+	mv -vif "${SRC}/amtool"                      "/bin/amtool" && \
+	mv -vif "${SRC}/alertmanager"                "/bin/alertmanager" && \
+	mv -vif "${SRC}/alertmanager.yml"            "/etc/alertmanager/alertmanager.yml" && \
+    chown -R nobody:nobody "/etc/alertmanager" "/alertmanager" && \
```
Comment (fabricetriboix) [inline Dockerfile]: Please use `root:root`

File: Dockerfile [hunk 5]
```diff
@@ -1 +1,32 @@
...
+	rm -rvf "${SRC}" package.tar.gz
+
+USER       nobody
```
Comment (fabricetriboix) [inline Dockerfile]: Please create a user using `useradd` and do not use the `nobody` user

File: Dockerfile [hunk 6]
```diff
@@ -1 +1,32 @@
...
+EXPOSE     9093
+VOLUME     [ "/alertmanager" ]
+WORKDIR    /alertmanager
```
Comment (fabricetriboix) [inline Dockerfile]: I find it rather odd that the WORKDIR is volume... Is this intended?

File: Dockerfile [hunk 7]
```diff
@@ -1 +1,32 @@
...
+ENTRYPOINT [ "/bin/alertmanager" ]
```
Comment (fabricetriboix) [inline Dockerfile]: Please do not use `ENTRYPOINT`, only use `CMD`

**Review by fabricetriboix — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 8]
```diff
@@ -1,32 +1,39 @@
...
+	mkdir -pv "/app/conf" && \
+	ln -sv "/app/conf" "/etc/alertmanager" && \
+	ln -sv "/app/conf" "/etc/amtool" && \
+	mv -vif "${SRC}/LICENSE"					"/LICENSE" && \
+	mv -vif "${SRC}/NOTICE"						"/NOTICE" && \
+	mv -vif "${SRC}/amtool"						"/bin/amtool" && \
+	mv -vif "${SRC}/alertmanager"				"/bin/alertmanager" && \
+	mv -vif "${SRC}/alertmanager.yml"			"/app/conf/alertmanager.yml" && \
+    chown -R nobody:nobody	"/app/data" "/app/conf" && \
+    chmod -R ug+rwX,o-rwx	"/app/data" "/app/conf" && \
```
Comment (fabricetriboix) [inline Dockerfile]: @drivera-armedia My point was to protect, say, `/app/conf` from the Prometheus process itself. The idea is that if the Prometheus process gets compromised, those files are still protected.

**Review by fabricetriboix — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 9]
```diff
@@ -1,39 +1,62 @@
...
-LABEL	ORG="Armedia LLC" \
-		APP="Prometheus Alert Manager" \
-		VERSION="${VER}" \
-		IMAGE_SOURCE="https://github.com/ArkCase/ark_prometheus_alertman" \
-		MAINTAINER="Armedia LLC"
+LABEL   ORG="Armedia LLC" \
+        APP="Prometheus Alert Manager" \
+        VERSION="${VER}" \
+        IMAGE_SOURCE="https://github.com/ArkCase/ark_prometheus_alertman" \
+        MAINTAINER="Armedia LLC"
```
Comment (fabricetriboix) [inline Dockerfile]: This is not consistent with your change in ark_prometheus...

**Review by fabricetriboix — DISMISSED** [review]

File: Dockerfile (hunk 9 above)
Comment (drivera-armedia) [reply Dockerfile]: You're right - something's off ... you're not seeing the latest versions. I'm positive I committed and pushed those yesterday.

**Review by fabricetriboix — APPROVED** [review]

**Review by JonHolman — APPROVED** [review]

**Review by JonHolman — APPROVED** [review]

**Pull request merged**